src/main/java/org/prebid/server/bidder/ix/IxBidder.java

```diff
@@ -111,74 +112,112 @@ private ExtImpIx parseImpExt(Imp imp) {
         }
     }
 
-    private static List<BidRequest> makeBidRequests(BidRequest bidRequest, Imp imp, String siteId) {
-        final Site modifiedSite = modifySite(bidRequest.getSite(), siteId);
+    private Imp modifyImp(Imp imp, ExtImpIx impExt) {
+        final UpdateResult<ObjectNode> impExtUpdateResult = modifyImpExt(imp, impExt);
+        final UpdateResult<Banner> bannerUpdateResult = modifyImpBanner(imp.getBanner());
 
-        return modifyImps(imp).stream()
-                .map(modifiedImp -> modifyBidRequest(bidRequest, modifiedSite, modifiedImp))
-                .toList();
+        return impExtUpdateResult.isUpdated() || bannerUpdateResult.isUpdated()
+                ? imp.toBuilder().ext(impExtUpdateResult.getValue()).banner(bannerUpdateResult.getValue()).build()
+                : imp;
     }
 
-    private static BidRequest modifyBidRequest(BidRequest bidRequest, Site site, Imp imp) {
-        return bidRequest.toBuilder()
-                .site(site)
-                .imp(Collections.singletonList(imp))
-                .build();
+    private UpdateResult<ObjectNode> modifyImpExt(Imp imp, ExtImpIx extImpIx) {
+        final String sid = extImpIx.getSid();
+        final ObjectNode impExt = imp.getExt();
+        if (StringUtils.isEmpty(sid)) {
+            return UpdateResult.unaltered(impExt);
+        }
+
+        final ObjectNode updatedExt = impExt.deepCopy();
+        updatedExt.set("sid", TextNode.valueOf(sid));
+
+        return UpdateResult.updated(updatedExt);
     }
 
-    private static Site modifySite(Site site, String siteId) {
-        return site == null
-                ? null
-                : site.toBuilder()
-                .publisher(modifyPublisher(site.getPublisher(), siteId))
-                .build();
+    private UpdateResult<Banner> modifyImpBanner(Banner banner) {
+        if (banner == null) {
+            return UpdateResult.unaltered(null);
+        }
+
+        final List<Format> formats = banner.getFormat();
+        final Integer w = banner.getW();
+        final Integer h = banner.getH();
+
+        if (formats == null || formats.isEmpty() && h != null && w != null) {
```

Comment:
`formats == null || formats.isEmpty()` can be replaced with `CollectionUtils.isEmpty()` for more clarity